test/setup.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+const chai = require('chai');
+
+global.sinon = require('sinon');
+global.assert = chai.assert;
```

Comment:
Instead of making this `require`s in the single test file, you've introduced a "complecated" setup scheme that is split into three parts (setup, config, the test itself).
Having test files in `test/lib/` is not a common practice of writing unit tests, even for projects from `nodules`.
Everything above is just a IMHO.